src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/AADFedAuthTokenRefreshTest/AADFedAuthTokenRefreshTest.cs

```diff
@@ -0,0 +1,133 @@
+            {
+                connection.Open();
+
+                // Set the token expiry to expire in 1 minute from now to force token refresh
+                string tokenHash1 = "";
+                DateTime? oldExpiry = GetOrSetTokenExpiryDateTime(connection, true, out tokenHash1);
+                Assert.True(oldExpiry != null, "Failed to make token expiry to expire in one minute.");
+
+                // Convert and display the old expiry into local time which should be in 1 minute from now
+                DateTime oldLocalExpiryTime = TimeZoneInfo.ConvertTimeFromUtc((DateTime)oldExpiry, TimeZoneInfo.Local);
+                LogInfo($"Token: {tokenHash1}   Old Expiry: {oldLocalExpiryTime}");
+                TimeSpan timeDiff = oldLocalExpiryTime - DateTime.Now;
+                Assert.True(timeDiff.TotalSeconds <= 60, "Failed to set expiry after 1 minute from current time.");
+
+                // Check if connection is still alive to continue further testing
+                string result = "";
+                SqlCommand cmd = connection.CreateCommand();
+                cmd.CommandText = "select @@version";
+                result = $"{cmd.ExecuteScalar()}";
+                Assert.True(result != string.Empty, "The connection's command must return a value");
+
+                // The new connection will use the same FedAuthToken but will refresh it first as it will expire in 1 minute.
+                using (SqlConnection connection2 = new SqlConnection(connStr))
+                {
+                    connection2.Open();
+
+                    // Check again if connection is alive
+                    cmd = connection2.CreateCommand();
+                    cmd.CommandText = "select 1";
+                    result = $"{cmd.ExecuteScalar()}";
+                    Assert.True(result != string.Empty, "The connection's command must return a value after a token refresh.");
+
+                    // Get the refreshed token expiry
+                    string tokenHash2 = "";
+                    DateTime? newExpiry = GetOrSetTokenExpiryDateTime(connection2, false, out tokenHash2);
+                    // Display new expiry in local time
+                    DateTime newLocalExpiryTime = TimeZoneInfo.ConvertTimeFromUtc((DateTime)newExpiry, TimeZoneInfo.Local);
+                    LogInfo($"Token: {tokenHash2}   New Expiry: {newLocalExpiryTime}");
+
+                    Assert.True(tokenHash1 == tokenHash2, "The token's hash before and after token refresh must be identical.");
+                    Assert.True(newLocalExpiryTime > oldLocalExpiryTime, "The refreshed token must have a new or later expiry time.");
+                }
+            }
+        }
+
+        private void LogInfo(string message)
+        {
+            _testOutputHelper.WriteLine(message);
+        }
+
+        private DateTime? GetOrSetTokenExpiryDateTime(SqlConnection connection, bool setExpiry, out string tokenHash)
```

Comment:
One of the drawbacks of using reflection is that it's not easy to maintain. Therefore, if you have to use it, consider preparing something more clear and usable, such as encapsulating it inside a helper class. You can find some samples of this approach in the same project.

Reply:
A helper class, FederatedAuthTokenHelper, was created and utilized.